crates/exex/src/manager.rs

```diff
@@ -0,0 +1,101 @@
+use std::{
+    collections::VecDeque,
+    future::Future,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+use reth_primitives::BlockNumber;
+use reth_provider::{CanonStateNotification, CanonStateNotifications};
+use tokio::sync::{mpsc::UnboundedSender, watch};
+use tokio_util::sync::PollSender;
+
+// todo
+// - exex height
+// - exex channel size
+// - manager buffer metrics
+// - throughput
+struct ExExMetrics;
+
+/// The execution extension manager.
+///
+/// The manager is responsible for:
+///
+/// - Receiving relevant events from the rest of the node, and sending these to the execution
+/// extensions
+/// - Backpressure
+/// - Error handling
+/// - Monitoring
+///
+/// TBD
+struct ExExManager {
+    // todo: exex receiver - get exex events
+    /// Channels from the manager to execution extensions.
+    exex_tx: Vec<PollSender<CanonStateNotification>>,
+
+    /// [`CanonStateNotification`] channel from the execution stage.
+    exec_rx: (), //UnboundedReceiverStream<CanonStateNotification>,
+    /// [`CanonStateNotification`] channel from the blockchain tree.
+    tree_rx: CanonStateNotifications,
+
+    /// Internal buffer of [`CanonStateNotification`]s.
+    buffer: VecDeque<CanonStateNotification>,
+    /// Max internal buffer size.
+    max_capacity: usize,
+    /// Current buffer capacity.
+    ///
+    /// Used to inform the execution stage of possible batch sizes.
+    current_capacity: Arc<AtomicUsize>,
+
+    /// Whether the manager is ready to receive new notifications.
+    is_ready: watch::Sender<bool>,
```

Comment:
nit: could be `Arc<AtomicBool>` ?

Reply:
has to be a watch channel since we want to wait for this to be true in `Stage::poll_ready` of the execution stage